qlkube/src/decorateSubscription.js

```diff
@@ -22,6 +22,199 @@ function getSubQueryList(queryName) {
   else return `${queryName}List`;
 }
 
+function getSublabels(targetType) {
+  return wrappers
+    .map(wtype => {
+      if (wtype['parents'].includes(targetType)) return wtype['type'];
+    })
+    .filter(s => {
+      return s;
+    });
+}
+
+function getFieldWrapper(targetType) {
+  return wrappers
+    .map(wtype => {
+      if (wtype['parents'].includes(targetType)) return wtype['fieldWrapper'];
+    })
+    .filter(s => {
+      return s;
+    })
+    .map(s => {
+      return uncapitalizeType(s);
+    });
+}
+
+function getSubType(baseSchema) {
+  return baseSchema._typeMap.Subscription === undefined
+    ? 'type Subscription'
+    : 'extend type Subscription';
+}
+
+function getResourceApiMainType(fieldName) {
+  return subscriptions.filter(sub => {
+    return `${sub.type}Update` === fieldName;
+  })[0];
+}
+
+function checkWrappedSubscription(
+  isList,
+  fieldWrapper,
+  newApiObj,
+  payload,
+  variables,
+  fieldName
+) {
+  let resultCheck = false;
+  let found = false;
+  let obj;
+  let targetObjField;
+
+  fieldWrapper.forEach(fw => {
+    if (!resultCheck) {
+      /*
+       * If the subscription is 'List' version, you find a match
+       * for each wrapped type and item
+       */
+      if (isList) {
+        for (let item of newApiObj.items) {
+          /*
+           * If found is true means that the item on the list
+           * was found and the payload.apiObj was updated
+           */
+          if (!found) {
+            /*
+             * Checks if the published event is about the main type
+             */
+            graphqlLogger(
+              `[i] Checks if the published event is about the main type of the item: ${JSON.stringify(
+                item
+              )} (subscription: ${fieldName} with variables: ${JSON.stringify(
+                variables
+              )})`
+            );
+            resultCheck =
+              resultCheck ||
+              ((item.metadata.namespace ||
+                item.metadata.namespace ===
+                  payload.apiObj.metadata.namespace) &&
+                (item.metadata.name === undefined ||
+                  item.metadata.name === payload.apiObj.metadata.name));
```

Comment:
I would suggest making a small function (with lots of returns) to improve readability, also it could be recycled afterwards